docs/connectors/building.mdx

```diff
@@ -3,19 +3,89 @@ title: "Building Connectors"
 
-The easiest way to start implementing your own Conduit connector is by using the
-[Conduit connector template](https://github.com/ConduitIO/conduit-connector-template).
-It contains the basic project structure as well as some additional utilities
-like GitHub actions and a Makefile.
+## Initializing a Conduit Connector Project
+
+To begin the development of a custom Conduit connector, developers should initialize their project using the [Conduit connector template](https://github.com/ConduitIO/conduit-connector-template). This template streamlines the setup process by providing a fundamental project structure, along with utility configurations for GitHub actions and a Makefile.
+
+Included in the Conduit Connector Template are:
+
+- Base code for configuring the connector's source and destination.
+- Sample unit tests to validate connector functionality.
+- A preconfigured Makefile to assist with common build tasks.
+- GitHub Actions Workflows for continuous integration including building, testing, linting, and automated release creation upon tag push.
+- Dependabot configurations to automate dependency updates and minor version auto-merging.
+- Issue and Pull Request templates for consistent contribution practices.
+- A README template to guide project documentation.
+
+### Steps to Use the Connector Template
+
+1. On the repository's main page, select "Use this template".
+2. Provide the new repository details as prompted.
+3. Upon repository creation, clone it to your local development environment.
+4. Execute the `./setup.sh` script with the desired module name, e.g., `./setup.sh github.com/awesome-org/conduit-connector-file`.
+5. (Optional) Define code owners in the `CODEOWNERS` file.
+
+## Connector Development Lifecycle
+
+A Conduit Connector can serve as a source, destination, or both. Implement a source connector by defining a struct that satisfies the [sdk.Source](<https://pkg.go.dev/github.com/conduitio/conduit-connector-sdk#Source>) interface, or create a destination connector by creating a struct that satisfies [sdk.Destination](<https://pkg.go.dev/github.com/conduitio/conduit-connector-sdk#Destination>).
+
+### Connector Component Files
+
+### `spec.go`
+
+This file defines the public-facing characteristics of the connector such as its name and capabilities. Begin by updating placeholder values with the specific details of your connector.
+
+![spec.go](/img/spec-go.png)
+
+### `config.go`
+
+This file holds configuration parameters shared between source and destination, if applicable. If separate configuration is required for source and destination, distinct `config.go` files should be created for each.
+
+![config.go](/img/config-go.png)
+
+### `paramgen_src` and `paramgen_dst`
+
+These files are created using the [paramgen](https://github.com/ConduitIO/conduit-connector-sdk/tree/main/cmd/paramgen) tool, which generates code from the Config struct to provide a map of configuration parameters.
```

Comment:
I'd also mention that we recommend adding a `go:generate` comment alongside the configuration type, which runs paramgen ([like here](https://github.com/ConduitIO/conduit-connector-template/blob/0e895d3346b946f8eaae5da39be007653ffe45a6/source.go#L3)). The makefile already contains the target `make generate`, which will execute all `go:generate` commands.

Reply:
@lovromazgon Added recommendation with ac7ddc6aa5811d4c26e192961df5a9c9d33eee7a